app/src/main/java/com/sdpteam/connectout/registration/RegistrationViewModel.java

```diff
@@ -0,0 +1,48 @@
+package com.sdpteam.connectout.registration;
+
+import static com.sdpteam.connectout.profile.Profile.Gender;
+import static com.sdpteam.connectout.registration.CompleteRegistration.MandatoryFields;
+
+import com.sdpteam.connectout.authentication.Authentication;
+import com.sdpteam.connectout.authentication.GoogleAuth;
+import com.sdpteam.connectout.profile.ProfileFirebaseDataSource;
+
+import androidx.lifecycle.ViewModel;
+
+public class RegistrationViewModel extends ViewModel {
+
+    private CompleteRegistration registration;
+    private Authentication auth;
+
+    public RegistrationViewModel() {
+        final ProfileFirebaseDataSource firebaseProfiles = new ProfileFirebaseDataSource();
+        registration = new CompleteRegistration(firebaseProfiles);
+        auth = new GoogleAuth();
+    }
+
+    String currentName() {
+        return auth.loggedUser().name;
+    }
+
+    String currentEmail() {
+        return auth.loggedUser().email;
+    }
+
+    public void completeRegistration(String name, String email, String bio, Gender g) {
+        if (auth.isLoggedIn()) {
+            registration.completeRegistration(auth.loggedUser().uid, new MandatoryFields(name, email, bio, g));
+        } else {
+            throw new IllegalStateException("Cannot complete the registration you're not even logged in.");
+        }
+    }
+
+    // for testing (mocking)
+    public void setRegistration(CompleteRegistration registration) {
```

Comment:
wouldn't it be better to inject this into a constructor instead of setting a public method just for this ? Wouldn't this go against the principles we learned in object oriented programming ?

Reply:
Oh yes nice catch forgot that I can pass arguments with the `ViewModelProvider.Factory` I discovered the other day.